### PR TITLE
O3-1840: Write E2E Test for Active Visit

### PR DESCRIPTION
**Pull request by anjula-sack**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

Created the following E2E Tests for active visit

- Active visits
- Visit note test

Related Doc: https://docs.google.com/document/d/1qD_zpYUG8C28zoAhrl7slI55lInyO7a-QR3tiwhrjwk/edit#heading=h.swr4izgpg4my

## Screenshots

image

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

File: e2e/commands/encounterOperations.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { Encounter } from './../../packages/esm-active-visits-app/src/visits-summary/visit.resource';
```
Comment (anjula-sack) [inline e2e/commands/encounterOperations.ts:1]: I imported this from a esm module. Is that okay? @jayasanka-sack

File: e2e/specs/activeVisits.spec.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  await homePage.clickOnActiveVisitPatient(patient.uuid);
+
+  await homePage.clickOnVisitSummaryTab();
+  await expect(page.getByRole('tabpanel', { name: encounterNote }).getByText('note')).toBeTruthy();
+});
+
+test.afterEach(async ({ api }) => {
+  await deletePatient(api, patient.uuid);
+  await endVisit(api, patient.uuid);
```
Comment (jayasanka-sack) [inline e2e/specs/activeVisits.spec.ts:50]:
```suggestion
  await endVisit(api, patient.uuid);
  await deletePatient(api, patient.uuid);
```

File: e2e/specs/activeVisits.spec.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+test('should be able to see the active visits', async ({ page }) => {
+  const homePage = new HomePage(page);
+  await homePage.goto();
+
+  await homePage.clickOnActiveVisitPatient(patient.uuid);
+
+  // Checks for the encounter
+  await expect(
+    page.getByTestId('encountersTable').getByRole('cell', { name: encounter.encounterType.display }),
+  ).toBeTruthy();
```
Comment (jayasanka-sack) [inline e2e/specs/activeVisits.spec.ts:28]: Can't we check for a value?

File: e2e/specs/patientSearch.spec.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ test.beforeEach(async ({ api }) => {
   patient = await generateRandomPatient(api);
 });
 
-test('should be able to search patients by identifier', async ({ page, api }) => {
+test('should be able to search patients by identifier', async ({ page }) => {
```
Comment (jayasanka-sack) [inline e2e/specs/patientSearch.spec.ts:12]: Good catch!

**Review by jayasanka-sack — COMMENTED** [review]
Thanks for the PR @anjula-sack ! Added some comments. Please check and let me know. Ping me if you need to have a huddle.

**Comment by jayasanka-sack** [comment]
The start command is updated. Please take a pull

File: e2e/specs/patientSearch.spec.ts (hunk 4 above)
Comment (anjula-sack) [reply e2e/specs/patientSearch.spec.ts:12]: eslint rule is not enforced. Found inside the esm modules files as well.

File: e2e/specs/activeVisits.spec.ts (hunk 3 above)
Comment (anjula-sack) [reply e2e/specs/activeVisits.spec.ts:28]: then each cell require unique id. This is pretty much the same. we are checking the created value to exist.

File: e2e/specs/activeVisits.spec.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  await expect(page.getByTestId(`${encounter.uuid}:provider`)).toContainText('Super User: Clinician');
+
+  // Checks for the visit note
+  await homePage.clickOnVisitSummaryTab();
+  await expect(page.getByTestId('note')).toContainText(encounterNote);
+});
+
+test.afterEach(async ({ api }) => {
+  await endVisit(api, patient.uuid);
+  await deletePatient(api, patient.uuid);
```
Comment (Piumal1999) [inline e2e/specs/activeVisits.spec.ts:45]: Does the created encounter get deleted when the patient is deleted?

Comment (anjula-sack) [reply e2e/specs/activeVisits.spec.ts:54]: No, Need to delete that as well. Thanks for pointing it out.

**Review by jayasanka-sack — APPROVED** [review]
Thanks @anjula-sack !

**Pull request merged**